File: mobile/modules/cjmux-vision-device/src/CJMUXVisionDeviceModule.ts
import { requireOptionalNativeModule } from "expo";
import type { CJMUXVisionDeviceModule } from "./CJMUXVisionDevice.types";

const fallback: CJMUXVisionDeviceModule = {
  isIOSAppOnVision: null,
} as CJMUXVisionDeviceModule;

export default requireOptionalNativeModule<CJMUXVisionDeviceModule>("CJMUXVisionDevice") ?? fallback;

File: mobile/modules/cjmux-vision-device/ios/CJMUXVisionDeviceModule.swift
import ExpoModulesCore
import Foundation

public class CJMUXVisionDeviceModule: Module {
  public func definition() -> ModuleDefinition {
    Name("CJMUXVisionDevice")

    Constant("isIOSAppOnVision") { () -> Bool? in
      if #available(iOS 26.1, *) {
        return ProcessInfo.processInfo.isiOSAppOnVision
      }
      return nil
    }
  }
}
